test/builtin_dataset_mocks.py

```diff
@@ -596,16 +608,27 @@ def generate(
         cls,
         root,
         *,
+        split,
         year,
         num_samples,
     ):
         annotations_dir = root / "annotations"
         annotations_dir.mkdir()
 
-        for split in ("train", "val"):
-            config_name = f"{split}{year}"
+        for split_ in ("train", "val"):
+            config_name = f"{split_}{year}"
+
+            images_meta = cls._make_images_meta(num_samples=num_samples)
```

Comment:
It's a bit strange to create the meta dict and its associated file for both splits while only create the image folder for one split.
I understand the meta dict is supposed to contain meta for both train and val, but right now the dict will contain meta for both splits while there will only be one image folder. Should we remove the `for loop` here?

Reply:
> I understand the meta dict is supposed to contain meta for both train and val, but right now the dict will contain meta for both splits while there will only be one image folder.

Unfortunately, that is how the dataset is structured:

```shell
$ ls
.rw-rw-r-- 253M root   31 May  2021 annotations_trainval2014.zip
.rw-rw-r-- 253M root   19 Oct  2021 annotations_trainval2017.zip
.rw-rw-r--  14G root   11 May  2021 train2014.zip
.rw-rw-r--  19G root   31 Oct  2021 train2017.zip
.rw-rw-r-- 6.6G root   31 Oct  2021 val2014.zip
.rw-rw-r-- 816M root   31 Oct  2021 val2017.zip
$ unzip -l annotations_trainval2014.zip
Archive:  annotations_trainval2014.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
332556225  2017-09-01 20:56   annotations/instances_train2014.json
160682675  2017-09-01 20:57   annotations/instances_val2014.json
170733465  2017-09-01 20:59   annotations/person_keypoints_train2014.json
 81637509  2017-09-01 20:59   annotations/person_keypoints_val2014.json
 66782097  2017-09-01 20:59   annotations/captions_train2014.json
 32421077  2017-09-01 20:59   annotations/captions_val2014.json
---------                     -------
844813048                     6 files
```

Reply:
This is also why we only create the annotations archive after the loop, whereas the image archive is created inside the loop:

https://github.com/pytorch/vision/blob/f176fb0dfc302a4cc58c374e59d2af6898629a33/test/builtin_dataset_mocks.py#L605-L615

Reply:
OK, fair enough. Just for my own understanding: everythign would still work if we removed the for loop, right?

Reply:
Yes, the test will still pass, but as explained in our guidelines, it would still pass if we remove 

https://github.com/pytorch/vision/blob/f176fb0dfc302a4cc58c374e59d2af6898629a33/torchvision/prototype/datasets/_builtin/coco.py#L161

On the original data, removing this line would fail.